packages/deploy/src/client.ts

```diff
@@ -37,6 +42,30 @@ export async function getProject(
   }
 }
 
+export async function downloadSpec(
+  config: DeployConfig,
+  projectId: string
+): Promise<{ data: ProjectPayload | null }> {
+  let base = new URL(config.endpoint).origin; 
+  let url = new URL(base + `/download/yaml`);
+  url.searchParams.set('id', projectId);
```

Comment:
I think this would be cleaner as:
```
const url = new URL(`/download/yaml?id=${projectId}`, config.endpoint)
```
This will construct the URL properly for you from the endpoint (the signature is `new URL(path, base)`) and set the URL query properly.

Side note, as we're in control of this URL, shouldn't it be more like `/download/:projectId` with an `Accept: application/yaml` header? Defaulting to yaml?